packages/bigquery/src/index.js

```diff
@@ -779,91 +952,49 @@ BigQuery.prototype.job = function(id) {
  *   var rows = data[0];
  * });
  */
-BigQuery.prototype.query = function(options, callback) {
-  var self = this;
-
-  if (is.string(options)) {
-    options = {
-      query: options
+BigQuery.prototype.query = function(query, options, callback) {
+  if (is.string(query)) {
+    query = {
+      query: query
     };
   }
 
-  options = options || {};
-
-  var job = options.job;
-
-  var requestQuery = extend({}, options);
-  delete requestQuery.job;
-
-  if (job) {
-    // Get results of the query.
-    delete requestQuery.params;
-    delete requestQuery.query;
-
-    self.request({
-      uri: '/queries/' + job.id,
-      qs: requestQuery
-    }, responseHandler);
-
-    return;
+  if (is.fn(options)) {
+    callback = options;
+    options = null;
   }
 
-  if (options.params) {
-    options.useLegacySql = false;
-    options.parameterMode = is.array(options.params) ? 'positional' : 'named';
+  var reqOpts = extend({}, query);
 
-    if (options.parameterMode === 'named') {
-      options.queryParameters = [];
+  if (query.params) {
+    reqOpts.useLegacySql = false;
```

Comment:
Seems a little confusing to automatically set the syntax type here only if `query.params` is set. Some of the other languages are defaulting to standard SQL in all cases, unless it's explicitly set to legacy.

Reply:
This was legacy code, I've gone ahead and set the default value for `useLegacySql` to false.